### PR TITLE
Feature/hir 126 add companies

### PR DESCRIPTION
**Pull request by AlexeyShok**
HIR-126. First step for HIR-125. Added Company entity. The ContractManager has been replaced with a Manager. Changed ownership of contracts and accommodations from Manager to Company. Added CompanyId to Manager. Added access rights set to Manager. Changed a database scheme to fit a new concept.

### CODE REVIEW

File: HappyTravel.Hiroshima.DirectManager/RequestValidators/ManagerRegisterRequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public class ManagerRegisterRequestValidator : AbstractValidator<Manager>
+    {
+        public ManagerRegisterRequestValidator()
+        {
+            RuleFor(manager => manager.FirstName).NotEmpty();
+            RuleFor(manager => manager.LastName).NotEmpty();
+            RuleFor(manager => manager.Title).NotEmpty();
+            RuleFor(manager => manager.Phone).NotEmpty().SetValidator(new PhoneNumberValidator());
+            RuleFor(manager => manager.Fax).SetValidator(new PhoneNumberValidator())
```
[the remark this answers is not in the document]
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.DirectManager/RequestValidators/ManagerRegisterRequestValidator.cs:14]: No. Validation is performed only if the condition is met where in the next line:
.When(manager => !string.IsNullOrEmpty(manager.Fax));

File: HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs [hunk 2]
```diff
@@ -164,27 +184,31 @@ private void AddContractAccommodationRelation(ModelBuilder modelBuilder)
...
+                e.HasKey(m => m.Id);
+                e.Property(m => m.IdentityHash).IsRequired();
+                e.Property(m => m.Email).IsRequired();
+                e.Property(m => m.FirstName).IsRequired();
+                e.Property(m => m.LastName).IsRequired();
+                e.Property(m => m.Title).IsRequired();
+                e.Property(m => m.Position).IsRequired();
+                e.Property(m => m.Phone).IsRequired();
+                e.Property(m => m.Fax);
+                e.Property(m => m.Created).IsRequired().HasDefaultValueSql("now() at time zone 'utc'");
```
[the remark this answers is not in the document]
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs:201]: Apparently the defaults were set earlier several migrations ago. When adding a new field of type not null, you must set a default value to apply to existing records in the database. Creation time is set everywhere in the code. If necessary, I can remove the default values in the next migration.

File: HappyTravel.Hiroshima.DirectManager/Services/AccommodationManagementService.cs [hunk 3]
```diff
@@ -36,9 +36,10 @@ public class AccommodationManagementService : IAccommodationManagementService
 
         public Task<Result<List<Models.Responses.Accommodation>>> GetAccommodations(int contractId)
         {
-            return _contractManagerContext.GetContractManager()
-                .EnsureContractBelongsToContractManager(_dbContext, contractId)
-                .Map(contractManager => GetContractAccommodations())
+            return _managerContext.GetManager()
+                .GetCompany(_dbContext)
```
[the remark this answers is not in the document]
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.DirectManager/Services/AccommodationManagementService.cs:40]: Not in all requests call GetCompany(). On the other hand, GetManager() is a common method for all requests that checks the user's access rights to the system. Therefore, such a scheme was chosen.

**Comment by kirillta** [comment]
Will you resolve these conflicts?

**Review by mingazhev — COMMENTED** [review]
Could you fix the comments I've added to the next PR there (which applicable)?

**Review by kirillta — APPROVED** [review]
lgtm

File: HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs [hunk 4]
```diff
@@ -168,27 +188,42 @@ private void AddContractAccommodationRelation(ModelBuilder modelBuilder)
...
-                
-                e.Property(cm => cm.Created).IsRequired().HasDefaultValueSql("now() at time zone 'utc'");
-                e.Property(cm => cm.Updated).IsRequired().HasDefaultValueSql("now() at time zone 'utc'");
-                e.Property(cm => cm.IsActive).IsRequired().HasDefaultValue(false);
-                
-                e.HasIndex(cm => cm.IdentityHash).IsUnique();
-                e.HasIndex(cm => cm.Email).IsUnique();
+                e.ToTable("ManagerServiceSupplierRelations");
+                e.HasKey(mssr => new { mssr.ManagerId, mssr.ServiceSupplierId });
+                e.Property(mssr => mssr.ManagerPermissions).IsRequired().HasDefaultValue(ManagerPermissions.All);   // All permissions
```
Comment (oleg1std) [inline HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs:224]: I think a manager no need to have all permissions by default

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerContextService.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+namespace HappyTravel.Hiroshima.DirectManager.Services
+{
+    public class ManagerContextService : IManagerContextService
+    {
+        public ManagerContextService(DirectContractsDbContext dbContext, ITokenInfoAccessor tokenInfoAccessor)
+        {
+            _dbContext = dbContext;
+            _tokenInfoAccessor = tokenInfoAccessor;
+        }
+
```
Comment (oleg1std) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerContextService.cs:17]: An empty line is needed

File: HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs (hunk 4 above)
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs:224]: This is the condition of the task. All existing managers receive full rights. Default value will be disabled in future migrations.

**Pull request merged**